src/main/java/redis/clients/jedis/JedisFactory.java

```diff
@@ -26,7 +26,11 @@ public JedisFactory(final String host, final int port, final int connectionTimeo
     this.hostAndPort.set(new HostAndPort(host, port));
     this.connectionTimeout = connectionTimeout;
     this.soTimeout = soTimeout;
-    this.password = password;
+    if(password != null && ! password.trim().isEmpty()){
```

Comment:
One thing I'm thinking about is, redis-cli trims parameter, so when redis-server's password is set to "foobared", I can get OK with typing "auth foobared     ".
I'm wondering it is just a redis-cli's behavior or redis-server's password is trimmed.